### PR TITLE
Trunk-5225:Get list of OrderGroups by patient or encounter is missing in openmrs…

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/TRUNK-5225
Get list of OrderGroups by patient or encounter is missing in openmrs…

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 1]
```diff
@@ -63,6 +72,80 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFrommMultipleEncounters() {
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		Encounter existingEncounter2 = Context.getEncounterService().getEncounter(6);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: What is the use of this test?

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 2]
```diff
@@ -63,6 +72,80 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldGetOrderGroupsFromMultiplePatients() {
+
+		Patient existingPatient1 = Context.getPatientService().getPatient(7);
+		Patient existingPatient2 = Context.getPatientService().getPatient(2);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: What is the use of this test?

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java (hunk 1 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: the purpose of this test is to create instances of multiple encounters and they do exist in the existing code

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java (hunk 2 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: the purpose of this test is to create instances of multiple patients and they do exist in the existing code

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: Can you point me to any tests that you found doing the same?

Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: your right @dkayiwa the tests are not there, am going to remove them and only maintain the tests relating to getting order groups from a patient

**Comment by dkayiwa** [comment]
Can you include the ticket id in your commit message as recommended at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by gitcliff** [comment]
oh sorry about this, had forgotten it,,,
have fixed it

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 3]
```diff
@@ -63,6 +72,46 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFromAnEncounter() {
+		Encounter existingEncounter = Context.getEncounterService().getEncounter(4);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: What is the reason for appending `1` to the variable name?

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 4]
```diff
@@ -63,6 +72,46 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+	}
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldGetOrderGroupsGivenPatient() {
+
+		Patient existingPatient1 = Context.getPatientService().getPatient(8);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByPatient(existingPatient1);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: What is the reason for appending 1 to the variable name?

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 5]
```diff
@@ -63,6 +72,46 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: Is this empty line intentional?

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java (hunk 3 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: i have fixed it

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java (hunk 4 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: i have fixed it

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java (hunk 5 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: No, i have fixed it

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java [hunk 6]
```diff
@@ -63,6 +72,41 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
...
+		assertEquals(1, ordergroups.size());
+	}
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldGetOrderGroupsGivenPatient() {
+		Patient existingPatient = Context.getPatientService().getPatient(8);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByPatient(existingPatient);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: Why `ordergroups1`?

Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java]: i have fixed it

**Pull request merged**